app/routes/goal_routes.py

```diff
@@ -1 +1,107 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, abort, make_response, request, Response
+
+from app.models.goal import Goal
+from app.models.task import Task
+from ..db import db
+from datetime import datetime
+from app.routes.utilities_routes import *
+
+bp = Blueprint("goal_bp",__name__, url_prefix= "/goals")
+
+
+####################################################################
+######################### Create FUNCTIONS #########################
+####################################################################
+@bp.post("")
+def create_goal():
+    request_body = request.get_json()
+    
+    if not request_body or not request_body.get("title"):
+        return make_response({"details": "Invalid data: missing title"}, 400) 
+    
+    goal = create_model(Goal, request_body)
+    return make_response(goal, 201)
```

Comment:
With the `create_model` refactor, `create_model` is already creating an appropriate response, so we can return it as
```py
    return create_model(Goal, request_body)
```